instrumentation/github.com/aws/aws-sdk-go/service/s3/otels3/s3.go

```diff
@@ -0,0 +1,216 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package otels3
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3iface"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+)
+
+var instrumentationName = "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go/service/s3/otels3"
+
+type instrumentedS3 struct {
+	s3iface.S3API
+	tracer          trace.Tracer
+	meter           metric.Meter
+	propagators     propagation.TextMapPropagator
+	counters        *counters
+	recorders       *recorders
+	spanCorrelation bool
+}
+
+type counters struct {
+	operation metric.Int64Counter
+}
+
+type recorders struct {
+	operationDuration metric.Float64ValueRecorder
+}
+
+// PutObjectWithContext invokes the PutObjectWithContext function with tracing instrumented.
+func (s *instrumentedS3) PutObjectWithContext(ctx aws.Context, input *s3.PutObjectInput, opts ...request.Option) (*s3.PutObjectOutput, error) {
+	startTime := time.Now()
+	destination := aws.StringValue(input.Bucket)
+	attrs := createAttributes(destination, operationPutObject)
+
+	spanCtx, span := s.tracer.Start(
+		ctx,
+		fmt.Sprintf("%s.%s", destination, operationPutObject),
+		trace.WithSpanKind(trace.SpanKindClient),
+		trace.WithAttributes(attrs...),
+	)
+
+	output, err := s.S3API.PutObjectWithContext(ctx, input, opts...)
+	callReturnTime := trace.WithTimestamp(time.Now())
+	defer span.End(callReturnTime)
+
+	if err != nil {
+		attrs = append(attrs, labelStatusFailure)
+		span.SetAttributes(labelStatusFailure)
```

Comment:
This seems redundant. Adding an attribute of the form `status=...` and setting the span status seems like a likely way to have this information exported twice. Do we need to add the status attribute here?

This applies to all the other wrapped methods as well.

Reply:
Added the span status as a part of https://github.com/open-telemetry/opentelemetry-go-contrib/pull/416#discussion_r526341069. 

I think we'd like to keep the attribute so that the status is also tracked in the metrics.